clients/callback_client.go

```diff
@@ -46,8 +46,10 @@ func NewPeriodicCallbackClient(callbackInterval time.Duration) *PeriodicCallback
 func (pcc *PeriodicCallbackClient) Start() *PeriodicCallbackClient {
```

Comment:
Hmm just thought of something regarding this periodic callback client, which is how do we get its benefits without having "terminal" updates like success or error get delayed up to 10 seconds when they are ideally received ASAP.

IOW today if the processing finishes with a result or error, it will go through the same flow as regular progress updates and only get saved in memory, to be flushed when the background routine gets to it. For terminal updates on the transcoding tho I think we should make the update immediately, as we'd be just busy waiting for a timer to finish the job.

WDYT? Might need to be addressed on a separate issue anyway.